uber/site_sections/groups.py

```diff
@@ -95,27 +140,7 @@ def unapprove(self, session, id, action, email, convert=None, message=''):
         if action == 'waitlisted':
             group.status = c.WAITLISTED
         else:
-            message = 'Group declined'
-            for attendee in group.attendees:
-                if attendee.is_unassigned or not convert:
-                    session.delete(attendee)
-                else:
-                    message = 'Group declined and emails sent to attendees'
-                    attendee.paid = c.NOT_PAID
-                    attendee.badge_status = c.NEW_STATUS
-                    attendee.ribbon = remove_opt(attendee.ribbon_ints, c.DEALER_RIBBON)
-                    attendee.overridden_price = attendee.new_badge_cost
-                    try:
-                        send_email(c.REGDESK_EMAIL, attendee.email, 'Do you still want to come to {EVENT_NAME}?',
-                                   render('emails/dealers/badge_converted.html', {
-                                       'attendee': attendee,
-                                       'group': group
-                                   }), model=attendee)
-                    except:
-                        message = 'Group declined (but the emails could not be sent)'
-                    group.attendees.remove(attendee)
```

Comment:
Ah, haha, oh boy. I just found a silly bug that has probably been causing subtle problems for awhile now. We're doing:
```python
for attendee in group.attendees:
    # ...
    group.attendees.remove(attendee)
```

We're modifying the attendees list as we iterate over it, thus causing the loop to skip every other attendee. So we were probably dumping a bunch of unassigned badges that are "Paid for by Group" into the system every time a group was rejected.

Reply:
Oh jeez, I had no idea that would happen. :c Good catch!